pkg/sql/logictest/testdata/logic_test/fk

```diff
@@ -874,15 +874,22 @@ pkref_b  CREATE TABLE pkref_b (
 
 # See https://github.com/cockroachdb/cockroach/issues/20045
 statement ok
-CREATE TABLE self_x2 (
+CREATE TABLE test20045 (
   x STRING PRIMARY KEY
- ,y STRING UNIQUE REFERENCES self_x2(x)
- ,z STRING REFERENCES self_x2(y)
- ,INDEX(z)
```

Comment:
Why did you take out INDEX(z).

I'd actually not make the entire 20045 related change in this COMMIT